pkg/forwarder/transactions_file_storage.go

```diff
@@ -56,6 +56,7 @@ func newTransactionsFileStorage(
 // Serialize serializes transactions to the file system.
 func (s *transactionsFileStorage) Serialize(transactions []Transaction) error {
 	s.telemetry.addSerializeCount()
+	_, _ = s.serializer.GetBytesAndReset()
```

Comment:
Why are we trashing what's in the serializer ? Same here, if needed this would require a comment explaining why.